File: android/app/src/main/kotlin/com/goalmaker/app/application/auth/AuthResult.kt
package com.goalmaker.app.application.auth

/** The outcome of a sign-in step, with failures the UI can explain in plain words. */
sealed interface AuthResult {
    data object Success : AuthResult

    data object WrongOrExpiredCode : AuthResult

    data object TooManyRequests : AuthResult

    data object Offline : AuthResult

    data class Failed(val detail: String) : AuthResult
}

File: windows/src/GoalMaker.Core/Auth/AuthResult.cs
namespace GoalMaker.Core.Auth;

/// <summary>The outcome of a sign-in step, with failures the UI can explain in plain words.</summary>
public abstract record AuthResult
{
    private AuthResult()
    {
    }

    public sealed record Success : AuthResult;

    public sealed record WrongOrExpiredCode : AuthResult;

    public sealed record TooManyRequests : AuthResult;

    public sealed record Offline : AuthResult;

    public sealed record Failed(string Detail) : AuthResult;
}
